lib/checkleakautovar.cpp

```diff
@@ -342,9 +349,19 @@ void CheckLeakAutoVar::checkScope(const Token * const startToken,
         while (Token::Match(ftok, "%name% :: %name%"))
             ftok = ftok->tokAt(2);
 
+        auto isAssignment = [](const Token* varTok) -> const Token* {
+            if (Token::Match(varTok, "%var% ="))
+                return varTok->next();
+            if (varTok->varId() && !Token::simpleMatch(varTok->astParent(), "[")) {
+                const Token* top = varTok->astTop();
+                if (top->str() == "=" && succeeds(top, varTok))
```

Comment:
sorry but I still fear some potential problems.

    buf[dostuff(123,p)] = ...

I guess it will determine that `p` is assigned.